httpie/cli/argtypes.py

```diff
@@ -225,7 +226,11 @@ def parse_format_options(s: str, defaults: Optional[dict]) -> dict:
                     f'invalid key {path!r}')
 
             default_type, parsed_type = type(default_value), type(parsed_value)
-            if parsed_type is not default_type:
+            if default_value is None:
+                # It is allowed to change an option when its default
+                # value is 'null' (parsed as `None`).
+                pass
```

Comment:
This makes me think that maybe using `''` as the default value could be cleaner. What do you think?

Reply:
:thinking: 

Using nothing:
```
  --format-options FORMAT_OPTIONS
      (...)
      
          headers.sort:true
          json.format:true
          json.indent:4
          json.sort_keys:true
          response.content_type:
          xml.format:true
          xml.indent:2
```

Using `''`:
```
  --format-options FORMAT_OPTIONS
      (...)
      
          headers.sort:true
          json.format:true
          json.indent:4
          json.sort_keys:true
          response.content_type:''
          xml.format:true
          xml.indent:2
```

Using `null`:
```
  --format-options FORMAT_OPTIONS
      (...)
      
          headers.sort:true
          json.format:true
          json.indent:4
          json.sort_keys:true
          response.content_type:null
          xml.format:true
          xml.indent:2
```

`''` has the advantage of the type check being the same (always `str`). And actually it renders better,  I think. `null` seems like a bad value not well translated.

Let's go with `''` :+1: